cmd/cloud-run/signifysecretrotator/signify/client.py

```diff
@@ -0,0 +1,123 @@
+"""Custom client for signify API"""
+
+import json
+import tempfile
+import requests
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography import x509
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.serialization import pkcs7
+
+
+class SignifyClient:
+    """Wraps signify API"""
+
+    def __init__(
+        self, token_url: str, certificate_service_url: str, client_id: str
+    ) -> None:
+        self.token_url = token_url
+        self.certificate_service_url = certificate_service_url
+        self.client_id = client_id
+
+    def fetch_access_token(self, certificate: bytes, private_key: bytes) -> str:
+        """fetches access token from given token_url using certificate and private key"""
+        # Use temporary file for old cert and key because requests library needs file paths,
+        # the code is running in known environment controlled by us
+        with (
+            tempfile.NamedTemporaryFile() as old_cert_file,
+            tempfile.NamedTemporaryFile() as old_key_file,
+        ):
+
+            old_cert_file.write(certificate)
+            old_cert_file.flush()
+
+            old_key_file.write(private_key)
+            old_key_file.flush()
+
+            access_token_response = requests.post(
+                self.token_url,
+                cert=(old_cert_file.name, old_key_file.name),
+                data={
+                    "grant_type": "client_credentials",
```

Comment:
Magic string

Reply:
It's grant_type, changing that will require change of method implementation. 
I prefer to keep it that way.

Reply:
Am I miss sth? "client_credentials" is a regular string

Reply:
It's regualr string, but it descripes grant type, which is static.
We can't switch from client_credentials to authorizaiton_code without changing the implementation of method.

Also this string is well defined in oAuth standards and everyone seeing it, should know exactly what that mean.
The best what I can is create enum, but I don't see the worth in it.